src/sqlancer/hive/HiveSchema.java

```diff
@@ -0,0 +1,114 @@
+package sqlancer.hive;
+
+import sqlancer.Randomly;
+import sqlancer.SQLConnection;
+import sqlancer.common.DBMSCommon;
+import sqlancer.common.schema.AbstractRelationalTable;
+import sqlancer.common.schema.AbstractSchema;
+import sqlancer.common.schema.AbstractTableColumn;
+import sqlancer.common.schema.AbstractTables;
+import sqlancer.common.schema.TableIndex;
+import sqlancer.hive.HiveSchema.HiveTable;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class HiveSchema extends AbstractSchema<HiveGlobalState, HiveTable> {
+
+    public enum HiveDataType {
+
+        // TODO: support more types, e.g. TIMESTAMP, DATE, VARCHAR, CHAR, BINARY, ARRAY, MAP, STRUCT, UNIONTYPE...
+        INT, FLOAT, DECIMAL, STRING, BOOLEAN;
+
+        public static HiveDataType getRandomType() {
+            return Randomly.fromList(Arrays.asList(values()));
+        }
+    }
+
+    public static class HiveColumn extends AbstractTableColumn<HiveTable, HiveDataType> {
+
+        public HiveColumn(String name, HiveTable table, HiveDataType type) {
+            super(name, table, type);
+        }
+    }
+
+    public static class HiveTables extends AbstractTables<HiveTable, HiveColumn> {
+
+        public HiveTables(List<HiveTable> tables) {
+            super(tables);
+        }
+    }
+
+    public static class HiveTable extends AbstractRelationalTable<HiveColumn, TableIndex, HiveGlobalState> {
+
+        public HiveTable(String name, List<HiveColumn> columns, boolean isView) {
+            super(name, columns, Collections.emptyList(), isView);
+        }
+    }
+
+    public HiveSchema(List<HiveTable> databaseTables) {
+        super(databaseTables);
+    }
+
+    public static HiveSchema fromConnection(SQLConnection con, String databaseName) throws SQLException {
+        // TODO: discover built-in tables
+        List<HiveTable> databaseTables = new ArrayList<>();
+        List<String> tableNames = getTableNames(con);
+        for (String tableName : tableNames) {
+            if (DBMSCommon.matchesIndexName(tableName)) {
```

Comment:
I think this might be a copy of DuckDB, or did you encounter a table name that matches an index during testing?

Reply:
If not, we should remove the if block.